### PR TITLE
Trinotate: update to 3.1.1 + fixes

### LINKED ISSUE #1770: Trinotate: Pre-generated Resource SQLite database not found
The download of the Trinotate Pre-generated Resource SQLite database fails because it has moved (now here: https://data.broadinstitute.org/Trinity/__deprecated_trinotate_resources/Trinotate_v3_RESOURCES/Trinotate_v3.sqlite.gz)

A quick fix is to update the URL in trinotate.xml.

But it seems that the sqlite database can now be build using the trinotate tool Build_Trinotate_Boilerplate_SQLite_db.pl (see https://trinotate.github.io/#SequencesRequired). I haven't tried, I am not an expert of Trinotate, but if this tool is ok, should be changed when the trinotate package version will be updated in the requirements.

### PR DESCRIPTION
**Pull request by abretaud**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

So here's an update of trinotate to version 3.1.1
It also uses the new trinotate command to create the initial sqlite db (instead of downloading) (fixes #1770)
And it fixes some conda installation problems with the older trinotate packages

### CODE REVIEW

**Comment by abretaud** [comment]
Ok, it's failing probably due to network: https://travis-ci.org/galaxyproject/tools-iuc/jobs/411786059#L2430
I'll retry next week

File: tools/trinotate/trinotate.xml [hunk 1]
```diff
@@ -1,23 +1,22 @@
-<tool id="trinotate" name="Trinotate" version="3.0.1.0">
+<tool id="trinotate" name="Trinotate" version="3.1.1.0">
```
Comment (nsoranzo) [inline tools/trinotate/trinotate.xml]: Version can be just `3.1.1`

**Comment by nsoranzo** [comment]
I think the error is `PASV ... couldn't connect to 141.161.180.197 port 1034: No route to host`, which I don't think would go away retrying. Is it possible to download using http instead of ftp?

**Comment by abretaud** [comment]
Argh... The ftp.ebi.ac.uk mirror is accessible by http, I'll have to patch the conda package to change the url in the code

**Comment by abretaud** [comment]
Aah green!

**Pull request merged**